front/src/applications/editor/tools/switchEdition/useSwitchTypes.ts

```diff
@@ -15,30 +15,47 @@ const trackNodeTypeOrder = [
   'double_slip_switch',
 ];
 
+let switchTypesCache: Record<number, SwitchType[]> = {};
+
 export default function useSwitchTypes(infraID: number | undefined) {
-  const [data, setData] = useState<SwitchType[]>([]);
+  const [data, setData] = useState<SwitchType[]>(
+    !isNil(infraID) ? switchTypesCache[infraID] || [] : []
```

Comment:
I m not sure to understand, if true why adding a new condition to possibly set an empty array?

Reply:
Here is what this does:
1. `data` must always be an array
2. If `infraID` is nil, then `data` is initially set to `[]`
3. Else, we check if we have a value in the cache that has already been loaded. If we have some, we set `data` to this value (`switchTypesCache[infraID]`)
4. Else, we set `data` to `[]`

@Yohh Does that answer your question?

Reply:
yes thanks